File: frontend/src/review/object/EligibilityReview.ts
import type {
  AppealStatus,
  EligibilityReviewId,
  EligibilityReviewTarget,
  EntityId,
  IsoDateTime,
  PersonName,
  ReasonText,
  ResolutionText,
} from '@/shared/object/domain/DomainObjects'

export type EligibilityReview = {
  id: EligibilityReviewId
  target: EligibilityReviewTarget
  targetId: EntityId
  targetName: PersonName
  reason: ReasonText
  status: AppealStatus
  resolutionNote?: ResolutionText
  submittedAt: IsoDateTime
  reviewedAt?: IsoDateTime
}

File: backend/src/main/scala/domain/review/EligibilityReview.scala
package domain.review

import domain.shared.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import domain.shared.*

final case class EligibilityReview(
    id: EligibilityReviewId,
    target: EligibilityReviewTarget,
    targetId: EntityId,
    targetName: DisplayText,
    reason: ReasonText,
    status: AppealStatus,
    resolutionNote: Option[ResolutionText],
    submittedAt: IsoDateTime,
    reviewedAt: Option[IsoDateTime],
)
object EligibilityReview:
  given Encoder[EligibilityReview] = deriveEncoder
  given Decoder[EligibilityReview] = deriveDecoder
